src/Microsoft.VisualStudio.ProjectSystem.Managed/IO/SimpleFileWatcher.cs

```diff
@@ -1,23 +1,31 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
 
+using Microsoft.VisualStudio.FileWatch;
+using Microsoft.VisualStudio.ProjectSystem;
+
 namespace Microsoft.VisualStudio.IO;
 
 /// <summary>
 /// Simple wrapper around the FileSystemWatcher.
 /// </summary>
-internal sealed class SimpleFileWatcher : IDisposable
+[Export(typeof(IFileWatcherService))]
+[Order(Order.Lowest)]
+internal sealed class SimpleFileWatcher : IFileWatcherService
 {
-    private FileSystemWatcher? _fileWatcher;
-    private FileSystemEventHandler? _handler;
-    private RenamedEventHandler? _renameHandler;
+    private readonly FileSystemWatcher? _fileWatcher;
+    public event EventHandler<FileWatcherEventArgs> OnDidCreate;
+    public event EventHandler<FileWatcherEventArgs> OnDidChange;
+    public event EventHandler<FileWatcherEventArgs> OnDidDelete;
 
-    // For unit tests
     public SimpleFileWatcher()
     {
     }
 
-    public SimpleFileWatcher(string dirToWatch, bool includeSubDirs, NotifyFilters notifyFilters, string fileFilter,
-                             FileSystemEventHandler? handler, RenamedEventHandler? renameHandler)
+    public SimpleFileWatcher(
+        string dirToWatch,
+        bool includeSubDirs,
+        NotifyFilters notifyFilters,
+        string fileFilter)
```

Comment:
This approach isn't likely to work. MEF won't be able to call this constructor. Instead it'll use the default (parameterless) constructor, which don't instantiate the `_fileWatcher`, meaning that no watching will take place.

This watcher doesn't feel like something that should be a MEF service. Its construction requires a file path, which isn't something that MEF would know about.